### PR TITLE
【Hackathon 5 No.21】Add LinearLR to Paddle

### PR DESCRIPTION
**Pull request by Asthestarsfalll**
### PR types

Others
### PR changes

APIs
### Description

Add LinearLR to Paddle

文档： https://github.com/PaddlePaddle/docs/pull/6219

### CODE REVIEW

File: python/paddle/optimizer/lr.py [hunk 1]
```diff
@@ -2229,6 +2230,125 @@ def get_lr(self):
...
+            ...                 'x': np.random.randn(3, 4, 5).astype('float32'),
+            ...                 'y': np.random.randn(3, 4, 5).astype('float32')
+            ...             },
+            ...             fetch_list=loss.name)
+            ...         scheduler.step()
+    """
+
+    def __init__(
+        self,
+        learning_rate,
```
Comment (jeff41404) [inline python/paddle/optimizer/lr.py:2306]: the parameter name needs to be consistent with the [rfc](https://github.com/PaddlePaddle/community/pull/647) and code, `base_learning_rate`in rfc, modify the document or modify the code.

Comment (Asthestarsfalll) [reply python/paddle/optimizer/lr.py:2306]: 已修改该rfc, PR: https://github.com/PaddlePaddle/community/pull/685

**Comment by Asthestarsfalll** [comment]
@jeff41404 PTAL

**Review by jeff41404 — DISMISSED** [review]
LGTM

File: python/paddle/optimizer/lr.py [hunk 2]
```diff
@@ -2229,6 +2230,125 @@ def get_lr(self):
...
+        end_factor (float) End learning rate is defined by `end_factor * learning_rate`. Default: 1.0.
+        last_epoch (int, optional): The index of last epoch. Can be set to restart training.Default: -1, means initial learning rate.
+        verbose: (bool, optional): If ``True``, prints a message to stdout for each update. Default: ``False`` .
+
+    Returns:
+        ``LinearLR`` instance to schedule learning rate.
+
+    Examples:
+        .. code-block:: python
+            :name: code-example1
```
Comment (sunzhongkai588) [inline python/paddle/optimizer/lr.py]: 建议 `name` 标识符号可以写清楚
```suggestion
            :name: code-dynamic
```

File: python/paddle/optimizer/lr.py [hunk 3]
```diff
@@ -2229,6 +2230,125 @@ def get_lr(self):
...
+            ...         x = paddle.uniform([10, 10])
+            ...         out = linear(x)
+            ...         loss = paddle.mean(out)
+            ...         loss.backward()
+            ...         sgd.step()
+            ...         sgd.clear_gradients()
+            ...         scheduler.step()
+
+        .. code-block:: python
+            :name: code-example2
```
Comment (sunzhongkai588) [inline python/paddle/optimizer/lr.py]:
```suggestion
            :name: code-static
```

**Review by sunzhongkai588 — COMMENTED** [review]
`name` 最好能改一下

File: python/paddle/optimizer/lr.py (hunk 2 above)
Comment (Asthestarsfalll) [reply python/paddle/optimizer/lr.py]: 已修改

**Review by sunzhongkai588 — APPROVED** [review]
LGTM for docs

**Pull request merged**